TaskManager/TaskManager/WeeklyTaskService.cs

```diff
@@ -0,0 +1,162 @@
+            Console.WriteLine("Input the priority number:");
+            var inputData = Console.ReadLine();
+            var priority = Enum.Parse<Priority>(inputData);
+            if (priority is >= 0 or < (Priority)4)
+            {
+                for (int counterOfPrintTask = 0, i = 0; i < _counter; i++)
+                {
+                    var task = _tasks[i];
+                    if (task.GetPriority() == priority)
+                    {
+                        PrintTask(task, i);
+                        counterOfPrintTask++;
+                    }
+                    if ((_counter - 1 == i) && (counterOfPrintTask == 0))
+                    {
+                        Console.WriteLine($"Tasks with priority {priority} not set");
+                    }
+                }
+            }
+            else
+            {
+                Console.WriteLine("Invalid priority number! Remember, you must only enter numbers from 0 to 3! \nTry again:");
+            }
+        }
+
+        private WeeklyTask ParseNewTask(string? inputData)
+        {
+            var parts = inputData?.Split(",");
+
+            if (parts == null || parts.Length < 1 || parts.Length > 4)
+            {
+                Console.WriteLine("Invalid task format, try again");
+                return null;
+            }
+
+            return CreateNewTask(parts);
+        }
+
+        private WeeklyTask CreateNewTask(string[] parts)
+        {
+            return parts.Length switch
+            {
+                1 => CreateTaskWithName(parts),
+                2 => CreateTaskWithDate(parts),
+                3 => CreateTaskWithDateTime(parts),
+                4 => CreateTaskWithDateAndPriority(parts),
+                _ => null,
+            };
+        }
+        
+        private WeeklyTask CreateTaskWithName(string[] parts) => new WeeklyTask(parts[0]);
+
+        private WeeklyTask CreateTaskWithDate(string[] parts)
+        {
+            var date = DateTime.Parse(parts[1]);
+            return new WeeklyTask(parts[0], date);
+        }
+        
+        private WeeklyTask CreateTaskWithDateTime(string[] parts)
+        {
+            var (date, time) = ParseDateTime(parts);
+            return new WeeklyTask(parts[0], date, time);
+        }
+
+        private WeeklyTask CreateTaskWithDateAndPriority(string[] parts)
+        {
+            var (date, time) = ParseDateTime(parts);
+
+            var priority = Enum.Parse<Priority>(parts[3]);
+            return new PriorityTask(parts[0], date, time, priority);
+        }
+
+        private static (DateTime date, DateTime time) ParseDateTime(string[] parts)
+        {
+            var date = DateTime.Parse(parts[1]);
+            var time = DateTime.Parse(parts[2]);
+            return (date, time);
+        }
+
+        private void AddNewTask(WeeklyTask task)
+        {
+            _tasks[_counter] = task;
+            _counter++;
+            Console.WriteLine("\nTask added successfully!\n");
```

Comment:
task service should no know about console, we'll improve this by adding a delegate (next homework)